README
This is the "facelift" datatracker branch that uses Twitter Bootstrap for
the UI.

You need to install a few new django extensions:
	https://pypi.python.org/pypi/django-widget-tweaks
	https://pypi.python.org/pypi/django-crispy-forms
	https://pypi.python.org/pypi/django-typogrify

The meta goal of this effort is: *** NO CHANGES TO THE PYTHON CODE ***

Whenever changes to the python code are made, they can only fix HTML bugs,
or add comments (tagged with "FACELIFT") about functionality that can be
removed once the facelift templates become default. Or they need to add
functionality that is only called from the new facelift templates.

Javascript that is only used on one template goes into that template.
Javascript that is used by more than one template goes into ietf.js.

CSS that is only used on one template goes into that template.
CSS that is used by more than one template goes into ietf.css. No CSS in the
templates or - god forbid - style tags! (And no CSS or HTML styling in
python code!!)

Templates that use jquery or bootstrap plugins include the css in the pagehead
block, and the js in the js block.


changelog
id2xml (1.5.2) ietf; urgency=medium

  * fix: Avoid category loop (#18)

 -- Kesara Rathnayake  <kesara@staff.ietf.org>  22 Jul 2023 19:32:44 +1200

id2xml (1.5.1) ietf; urgency=medium

  * test: Improve tests (#14)

  * fix: Parse boilerplate with revised BSD license (#13)

 -- Kesara Rathnayake  <kesara@staff.ietf.org>  12 Jul 2023 21:47:44 +1200

id2xml (1.5.0) ietf; urgency=medium

  * Fixed a few Python 3.x nits in the parser code.

  * Changed generated anchor slugs so as to not collide with IDs used by 
    xml2rfc, which in v3 mode reserves certain prefixes used for section, 
    figure and table anchors and fragment identifiers.  Also fixed an xml2rfc 
    v2v3 convertere compatibility issue.

  * Added outdent to id2xml progress statements in order to more clearly 
    differentiate them from warnings and errors.

  * Added the possibility of setting document stream, IPR, and consensus 
    values with switches instead of deriving them from the 'Status of this 
    Memo' section, to make processing of documents with old status wording 
    easier.

  * Added a list of valid submission types and stream names to id2xml.utils.

  * Tweaked several command-line options to better match those of xml2rfc.  
    Added help for trace options when --help is given together with --debug.

 -- Henrik Levkowetz <henrik@levkowetz.com>  03 Sep 2019 12:30:14 +0000


id2xml (1.4.4) ietf; urgency=medium

  * Since '--' isn't allowed inside html comments, trying to add a comment 
    that contains that string will fail.  Under some circumstances, id2xml
    inserts comments about conversion errors into the generated xml, and
    if the failed text contained '--', this would fail.  Tweaked comment
    insertion to avoid this.

  * Added code to look for Abstract both before and after Status-of-Memo 
    and Copyright sections.  The placement of the Abstract changes around the 
    time of RFC 5550.

 -- Henrik Levkowetz <henrik@levkowetz.com>  17 Oct 2018 12:10:41 +0000

id2xml (1.4.3) ietf; urgency=medium

  * Fixed some issues with author recognition and an infinite loop.

  * Added handling for the grammar change 'a candidate' --> 'candidates' in 
    the status of memo boilerplate.

  * Aligned the short options for verbose and version with common practice.

 -- Henrik Levkowetz <henrik@levkowetz.com>  12 Oct 2018 14:05:46 +0000


id2xml (1.4.2) ietf; urgency=medium

  * Fixed a couple of issues that prevented processing of rfc7752.

  * Aligned generator comment date format with that used by xml2rfc

  * Added a 'Generated by' comment showing id2xml version and time of run.

 -- Henrik Levkowetz <henrik@levkowetz.com>  19 Jul 2018 14:33:25 +0000


id2xml (1.4.1) ietf; urgency=medium

  This release which provides 2 small enhancements:

  * Deal better with comma-separated lists of rfc numbers in Updates: and 
    Obsoletes: fields in the input document header.

  * Added warnings as comments inline in the xml file (in addition to 
    showing them on stderr).

 -- Henrik Levkowetz <henrik@levkowetz.com>  19 Dec 2017 22:39:48 +0000

id2xml (1.4.0) ietf; urgency=medium

  This is a small feature release which adds support for tables with
  style='all'.

 -- Henrik Levkowetz <henrik@levkowetz.com>  02 Nov 2017 18:06:11 +0000


id2xml (1.3.1) ietf; urgency=medium

  * Added http:/https: normalization when comparing boilerplate.  Also 
    tweaked punctuation in some warning messages, for consistency.  Permitted 
    some additional variation in appendix section numbering.

  * Added a new test document, containing https:// boilerplate URLs.

  * Added Glen and Matt @amsl.com to the release-mail recipients.

 -- Henrik Levkowetz <henrik@levkowetz.com>  30 Oct 2017 13:10:19 +0000


id2xml (1.3.0) ietf; urgency=medium

  This release adds a --schema-v3 switch, which will cause id2xml to generate
  xml according to xml2rfc vocabulary vesion 3.  Earlier it was indicated that
  this would be available with version 2.0.0 of id2xml, but the v3 generation
  was accomplished with much less of a rewrite than anticipated, and the v2
  output is still the default, so a major version bump seems inappropriate.

  Details from the commit log:

  * Changed a dependency from pathlib to the recommended replacement 
    pathlib2, and added a version requirement to the xml2rfc dependency.

  * Added support for id2xml conversion to xml2rfc vocabulary version 3 
    format by using the v2v3 conversion component of xml2rfc.  This is invoked 
    with a new switch: -3 / --schema-v3.

 -- Henrik Levkowetz <henrik@levkowetz.com>  23 Sep 2017 15:47:13 +0000


id2xml (1.2.3) ietf; urgency=medium

  * Added support for using draft names in references to identify 
    references that may have bibxml entries, and use those.  If found, update 
    the xref targets in the document accordingly.  Updated baseline diffs.

 -- Henrik Levkowetz <henrik@levkowetz.com>  23 Sep 2017 12:27:53 +0000

id2xml (1.2.2) ietf; urgency=medium

  * Added a warning when an author name is seen in the document title 
    abbreviation.

 -- Henrik Levkowetz <henrik@levkowetz.com>  10 Aug 2017 15:09:39 +0000


id2xml (1.2.1) ietf; urgency=medium

  * Fixed a bug in the handling of unnumbered reference sections.

 -- Henrik Levkowetz <henrik@levkowetz.com>  10 Aug 2017 11:59:24 +0000


id2xml (1.2.0) ietf; urgency=medium

  This is a small feature release which adds some variations to the recognised
  title of the copyright section.  There's also a small bugfix:

  * Added recognition of additional variations on the title of the 
    copyright section, based on common variations received from the RFC Editor.

  * Fixed a bug in the handling of docname parts in reference text, when 
    the docname cannot be split into series info.

 -- Henrik Levkowetz <henrik@levkowetz.com>  09 Aug 2017 18:52:16 +0000


id2xml (1.1.0) ietf; urgency=medium

  This release provides minor changes in response to feedback from the
  RFC Editor staff.


sec/settings.py
# Feb 24 2012 

# Django settings for sec project.

import os
import syslog
syslog.openlog("django", syslog.LOG_PID, syslog.LOG_LOCAL0)

BASE_DIR = os.path.dirname(os.path.abspath(__file__))

#import sys
#sys.path.append(os.path.abspath(BASE_DIR + "/.."))
#sys.path.append(os.path.abspath(BASE_DIR + "/../redesign"))

DEBUG = False

# Domain name of the IETF
IETF_DOMAIN = 'ietf.org'

ADMINS = (
    ('Ryan Cross', 'rcross@amsl.com'),
)

# Server name of the tools server
TOOLS_SERVER = 'tools.' + IETF_DOMAIN

# Override this in the settings_local.py file:
#SERVER_EMAIL = 'Django Server <django-project@' + TOOLS_SERVER + '>'

DEFAULT_FROM_EMAIL = 'IETF Secretariat <ietf-secretariat-reply@' + IETF_DOMAIN + '>'

MANAGERS = ADMINS

# DATABASES defined in settings_local.py
 
# Local time zone for this installation. Choices can be found here:
# http://www.postgresql.org/docs/8.1/static/datetime-keywords.html#DATETIME-TIMEZONE-SET-TABLE
# although not all variations may be possible on all operating systems.
# If running in a Windows environment this must be set to the same as your
# system time zone.
TIME_ZONE = 'PST8PDT'

# Language code for this installation. All choices can be found here:
# http://www.w3.org/TR/REC-html40/struct/dirlang.html#langcodes
# http://blogs.law.harvard.edu/tech/stories/storyReader$15
LANGUAGE_CODE = 'en-us'

SITE_ID = 1

# If you set this to False, Django will make some optimizations so as not
# to load the internationalization machinery.
USE_I18N = False

# Absolute path to the directory that holds media.
# Example: "/home/media/media.lawrence.com/"
MEDIA_ROOT = '/a/www/www6s'

# URL that handles the media served from MEDIA_ROOT.
# Example: "http://media.lawrence.com"
MEDIA_URL = 'http://www.ietf.org'

# URL prefix for admin media -- CSS, JavaScript and images. Make sure to use a
# trailing slash.
# Examples: "http://foo.com/media/", "/media/".
ADMIN_MEDIA_PREFIX = '/media/'

#AUTH_PROFILE_MODULE = 'ietfauth.IetfUserProfile'
#AUTHENTICATION_BACKENDS = ( "ietf.ietfauth.auth.IetfUserBackend", )
AUTH_PROFILE_MODULE = 'person.Person'
AUTHENTICATION_BACKENDS = ( 'django.contrib.auth.backends.RemoteUserBackend', )

#DATABASE_ROUTERS = ["ietf.legacy_router.LegacyRouter"]

SESSION_COOKIE_AGE = 43200 # 12 hours
SESSION_EXPIRE_AT_BROWSER_CLOSE = True

TEMPLATE_LOADERS = (
    'django.template.loaders.filesystem.load_template_source',
    'django.template.loaders.app_directories.load_template_source',
)

MIDDLEWARE_CLASSES = (
    'django.middleware.common.CommonMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.auth.middleware.RemoteUserMiddleware',
    'django.middleware.doc.XViewMiddleware',
    'ietf.middleware.SQLLogMiddleware',
    'ietf.middleware.SMTPExceptionMiddleware',
    'ietf.middleware.RedirectTrailingPeriod',
    'django.middleware.transaction.TransactionMiddleware',
    'ietf.middleware.UnicodeNfkcNormalization',
    'sec.middleware.secauth.SecAuthMiddleware',
)

ROOT_URLCONF = 'sec.urls'

TEMPLATE_DIRS = (
    BASE_DIR + "/templates",
    "/a/www/ietf-datatracker/web/ietf/templates",
)

TEMPLATE_CONTEXT_PROCESSORS = (
    'django.core.context_processors.auth',
    'django.core.context_processors.debug',
    'django.core.context_processors.i18n',
    'django.core.context_processors.media',
    'django.core.context_processors.request',
    'django.contrib.messages.context_processors.messages',
    'sec.context_processors.server_mode',
    'sec.context_processors.revision_info',
)

INSTALLED_APPS = (
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.sites',
    'django.contrib.sitemaps',
    'django.contrib.admin',
    'django.contrib.admindocs',
    'django.contrib.humanize',
    'django.contrib.messages',
    'django.contrib.formtools',
    'south',
    'workflows',
    'permissions',
    #'ietf.issue', #this is bogus leave it out
    'ietf.announcements',
    'ietf.doc',
    'ietf.group',
    'ietf.idindex',
    'ietf.idrfc',
    'ietf.idtracker',
    'ietf.iesg',
    'ietf.ietfauth',
    'ietf.ietfworkflows',
    'ietf.ipr',
    'ietf.liaisons',
    'ietf.mailinglists',
    'ietf.meeting',
    'ietf.message',
    'ietf.name',
    'ietf.person',
    #'ietf.proceedings', # deprecated
    'ietf.redirects',
    'ietf.submit',
    'ietf.wgchairs',
    'ietf.wgcharter',
    'ietf.wginfo',
    # new apps
    'sec.announcement',
    'sec.areas',
    'sec.console',
    'sec.drafts',
    'sec.groups',
    'sec.ipr',
    'sec.meetings',
    'sec.proceedings',
    'sec.roles',
    'sec.rolodex',
    'sec.sreq',
    'sec.telechat',
    'form_utils',
    'django_extensions',
)

# INTERNAL_IPS undefined in production

# this is a tuple of regular expressions.  if the incoming URL matches one of
# these, than non secretariat access is allowed.
SEC_AUTH_UNRESTRICTED_URLS = (
    (r'^/$'),
    (r'^/announcement/'),
    (r'^/proceedings/'),
    (r'^/sreq/'),
)

# no slash at end
IDTRACKER_BASE_URL = "http://datatracker.ietf.org"

# Valid values:
# 'production', 'test', 'development'
# Override this in settings_local.py if it's not true
SERVER_MODE = 'production'

# WG Chair configuration
MAX_WG_DELEGATES = 3

# Override this in settings_local.py if needed
# *_PATH variables ends with a slash/ .
CONFLICT_REVIEW_PATH = '/a/www/ietf-ftp/conflict-reviews/'
INTERNET_DRAFT_PATH = '/a/www/ietf-ftp/internet-drafts/'
INTERNET_DRAFT_PDF_PATH = '/a/www/ietf-datatracker/pdf/'
RFC_PATH = '/a/www/ietf-ftp/rfc/'
AGENDA_PATH = '/a/www/www6s/proceedings/'
AGENDA_PATH_PATTERN = '/a/www/www6s/proceedings/%(meeting)s/agenda/%(wg)s.%(ext)s'
MINUTES_PATH_PATTERN = '/a/www/www6s/proceedings/%(meeting)s/minutes/%(wg)s.%(ext)s'
SLIDES_PATH_PATTERN = '/a/www/www6s/proceedings/%(meeting)s/slides/%(wg)s-*'
IPR_DOCUMENT_PATH = '/a/www/ietf-ftp/ietf/IPR/'
IETFWG_DESCRIPTIONS_PATH = '/a/www/www6s/wg-descriptions/'
IESG_TASK_FILE = '/a/www/www6/iesg/internal/task.txt'
IESG_ROLL_CALL_FILE = '/a/www/www6/iesg/internal/rollcall.txt'
IESG_MINUTES_FILE = '/a/www/www6/iesg/internal/minutes.txt'
IESG_WG_EVALUATION_DIR = "/a/www/www6/iesg/evaluation"
INTERNET_DRAFT_ARCHIVE_DIR = '/a/www/www6s/draft-archive'
INTERNET_DRAFT_DAYS_TO_EXPIRE = 185

# Override this in settings_local.py if needed
#CACHE_MIDDLEWARE_SECONDS = 300
#CACHE_MIDDLEWARE_KEY_PREFIX = ''
#if SERVER_MODE == 'production':
#    CACHE_BACKEND= 'file://'+'/a/www/ietf-datatracker/cache/'
#else:
    # Default to no caching in development/test, so that every developer
    # doesn't have to set CACHE_BACKEND in settings_local
#    CACHE_BACKEND = 'dummy:///'

IPR_EMAIL_TO = ['ietf-ipr@ietf.org', ]
DOC_APPROVAL_EMAIL_CC = ["RFC Editor <rfc-editor@rfc-editor.org>", ]

# Liaison Statement Tool settings
LIAISON_UNIVERSAL_FROM = 'Liaison Statement Management Tool <lsmt@' + IETF_DOMAIN + '>'
LIAISON_ATTACH_PATH = '/a/www/ietf-datatracker/documents/LIAISON/'
LIAISON_ATTACH_URL = '/documents/LIAISON/'

# ID Submission Tool settings
IDSUBMIT_FROM_EMAIL = 'IETF I-D Submission Tool <idsubmission@ietf.org>'
IDSUBMIT_TO_EMAIL = 'internet-drafts@ietf.org'
IDSUBMIT_ANNOUNCE_FROM_EMAIL = 'internet-drafts@ietf.org'
IDSUBMIT_ANNOUNCE_LIST_EMAIL = 'i-d-announce@ietf.org'

# Days from meeting to cut off dates on submit
FIRST_CUTOFF_DAYS = 20
SECOND_CUTOFF_DAYS = 13
CUTOFF_HOUR = 17
SUBMISSION_START_DAYS = -90
SUBMISSION_CUTOFF_DAYS = 33
SUBMISSION_CORRECTION_DAYS = 59

IDSUBMIT_REPOSITORY_PATH = INTERNET_DRAFT_PATH
IDSUBMIT_STAGING_PATH = '/a/www/www6s/staging/'
IDSUBMIT_STAGING_URL = 'http://www.ietf.org/staging/'
IDSUBMIT_IDNITS_BINARY = '/a/www/ietf-datatracker/scripts/idnits'

MAX_PLAIN_DRAFT_SIZE = 6291456  # Max size of the txt draft in bytes

# DOS THRESHOLDS PER DAY (Sizes are in MB)
MAX_SAME_DRAFT_NAME = 20
MAX_SAME_DRAFT_NAME_SIZE = 50
MAX_SAME_SUBMITTER = 50
MAX_SAME_SUBMITTER_SIZE = 150
MAX_SAME_WG_DRAFT = 150
MAX_SAME_WG_DRAFT_SIZE = 450
MAX_DAILY_SUBMISSION = 1000
MAX_DAILY_SUBMISSION_SIZE = 2000
# End of ID Submission Tool settings

# DB redesign
USE_DB_REDESIGN_PROXY_CLASSES = True

# no migrations
SOUTH_TESTS_MIGRATE = False

if USE_DB_REDESIGN_PROXY_CLASSES:
    AUTH_PROFILE_MODULE = 'person.Person'
    AUTHENTICATION_BACKENDS = ( 'django.contrib.auth.backends.RemoteUserBackend', )

CHARTER_PATH = '/a/www/ietf-ftp/charter/'
CHARTER_TXT_URL = 'http://www.ietf.org/charters/'

# AMS additions
# for meetings materials upload
AMS_STATICFILES = BASE_DIR + '/../static/'
BLUE_SHEET_PATH = '/a/www/ietf-datatracker/documents/blue_sheet.rtf'
BLUE_SHEET_URL = 'https://datatracker.ietf.org/documents/blue_sheet.rtf'
DEVELOPMENT = False
GROUP_DESCRIPTION_DIR = '/a/www/www6s/wg-descriptions'
INTERIM_LISTING_DIR = '/a/www/www6/meeting/interim'
MAX_UPLOAD_SIZE = 40960000
PROCEEDINGS_DIR = '/a/www/www6s/proceedings/'
AGENDA_PATH = '/a/www/www6s/proceedings/'   # for code merge

# Secretariat Tool
# this is a tuple of regular expressions.  if the incoming URL matches one of
# these, than non secretariat access is allowed.
SECR_AUTH_UNRESTRICTED_URLS = (
    #(r'^/$'),
    (r'^/secr/announcement/'),
    (r'^/secr/proceedings/'),
    (r'^/secr/sreq/'),
)
SECR_BLUE_SHEET_PATH = '/a/www/ietf-datatracker/documents/blue_sheet.rtf'
SECR_BLUE_SHEET_URL = 'https://datatracker.ietf.org/documents/blue_sheet.rtf'
SECR_INTERIM_LISTING_DIR = '/a/www/www6/meeting/interim'
SECR_MAX_UPLOAD_SIZE = 40960000
SECR_PROCEEDINGS_DIR = '/a/www/www6s/proceedings/'
SECR_STATIC_URL = '/secr-static/'

# Put SECRET_KEY in here, or any other sensitive or site-specific
# changes.  DO NOT commit settings_local.py to svn.
from settings_local import *
